### PR TITLE
[RW-3231][RISK=NO] Update Data set page

### PR DESCRIPTION
**Pull request by NehaBroad**
Changes : 1) Add sticky footer which includes Save( orUpdate) and Analyze button
2) Update the Preview data header :
a) To indicate it is step 4
b) Include the link 'View Preview Table' (which will always be enable)
c) Remove Refresh preview link

Screen Shot 2019-08-09 at 5 09 49 PM

Screen Shot 2019-08-09 at 5 10 32 PM

### CODE REVIEW

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 1]
```diff
@@ -719,15 +737,24 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
...
+        <div>
+          <div style={styles.footer} />
+          <div style={styles.stickyFooter}>
+            <TooltipTrigger data-test-id='save-tooltip'
+            content='Requires Owner or Writer permission' disabled={this.canWrite}>
+            <Button style={{marginBottom: '2rem'}} data-test-id='save-button'
+            onClick ={() => this.setState({openSaveModal: true})}
+            disabled={this.disableSave() || !this.canWrite}>
+            {this.editing ? !(dataSetTouched && this.canWrite) ? 'Analyze' :
+            'Update And Analyze' : 'Save And Analyze'}
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/dataset-page.tsx]: "And" should be lowercased here

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 2]
```diff
@@ -650,34 +677,25 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
...
-                <Button data-test-id='save-button'
-                  onClick ={() => this.setState({openSaveModal: true})}
-                  disabled={this.disableSave() || !this.canWrite}>
-                  {this.editing ? !(dataSetTouched && this.canWrite) ? 'Analyze' :
-                    'Update And Analyze' : 'Save And Analyze'}
-                </Button>
-              </TooltipTrigger>
+              <Clickable data-test-id='preview-button' style={{marginTop: '0.5rem',
+                height: '1.8rem', width: '6.5rem', color: colors.secondary}}
+                onClick={() => this.getPreviewList()}>
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/dataset-page.tsx]: I'm confused - what happens now if you click this without selecting anything? It shows an error modal? This seems worse than the current behavior

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 3]
```diff
@@ -719,15 +737,24 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
...
-                </Button>
               </div>
             }
           </div>
         </FadeBox>
+        <div>
+          <div style={styles.footer} />
+          <div style={styles.stickyFooter}>
+            <TooltipTrigger data-test-id='save-tooltip'
+            content='Requires Owner or Writer permission' disabled={this.canWrite}>
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/dataset-page.tsx]: indent?

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 4]
```diff
@@ -719,15 +737,24 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
...
+            <Button style={{marginBottom: '2rem'}} data-test-id='save-button'
+            onClick ={() => this.setState({openSaveModal: true})}
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/dataset-page.tsx]: indent?

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 5]
```diff
@@ -719,15 +737,24 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
...
+            disabled={this.disableSave() || !this.canWrite}>
+            {this.editing ? !(dataSetTouched && this.canWrite) ? 'Analyze' :
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/dataset-page.tsx]: indent?

File: ui/src/app/pages/data/data-set/dataset-page.tsx (hunk 1 above)
Comment (calbach) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: Also in the mocks it uses all caps on the button, though that is probably best implemented via CSS if we do that.

Comment (NehaBroad) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: button change text to all Upper case i can change this just for readability

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 6]
```diff
@@ -650,34 +673,28 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
               <div style={{display: 'flex', flexDirection: 'column'}}>
               <div style={{display: 'flex', alignItems: 'flex-end'}}>
                 <div style={styles.previewDataHeader}>
-                  Preview Data Set
+                  <div>
+                    <CircleWithText text={4} width='23.78px' height='23.78px'
+                      style={{marignTop: '0.3rem', fill: colorWithWhiteness(colors.primary, 0.5)}}/>
+                  </div>
+                  <label style={{marginLeft: '0.5rem', color: colors.primary}}>
+                    Preview Data Set
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/dataset-page.tsx]: Existing issue, but the mocks say "Dataset" in most places, this should match. This file is not self-consistent on this

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 7]
```diff
@@ -719,15 +736,24 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
...
-                        style={{marginTop: '0.5rem', height: '1.8rem', width: '6.5rem'}}
-                        onClick={() => this.getPreviewList()}>
-                  Preview Table
-                </Button>
               </div>
             }
           </div>
         </FadeBox>
+        <div>
+          <div style={styles.footer} />
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/dataset-page.tsx]: I haven't patched this locally but please test this very carefully locally. Sticky footers are very difficult to get right in my experience.

Please try alternate window sizes when testing.

**Pull request merged**